### PR TITLE
[Annoncements] Little Announce callbacks improvement

### PR DESCRIPTION
**Pull request by CastagnaIT**
## Description

Save little memory/cpu consumption

## Motivation and Context

The callbacks concerning "annuncements" can be done hundreds of times and more over the time by Kodi or all the installed add-ons.
For the particular case of JSON TCPServer, there is a useless overhead, this feature is most of the time never used,
and there is the JSONRPC data conversion that happen at every callback that can be heavy for low-end devices when there is big data.

## How Has This Been Tested?

locally by calling from python addon `xbmc.executeJSONRPC` `JSONRPC.NotifyAll`

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/network/TCPServer.cpp [hunk 1]
```diff
@@ -227,6 +227,9 @@ void CTCPServer::Announce(ANNOUNCEMENT::AnnouncementFlag flag,
                           const std::string& message,
                           const CVariant& data)
 {
+  if (!m_connections.size())
```
Comment (enen92) [inline xbmc/network/TCPServer.cpp:230]: Is this threadsafe? A bit below while iterating m_connections Kodi obtains the lock

Comment (CastagnaIT) [reply xbmc/network/TCPServer.cpp:230]: but is within the loop the size is read before,
to make things as before i could put the size value in to a new variable and use it also in the loop

Comment (enen92) [reply xbmc/network/TCPServer.cpp:230]: Disregard, lock is in the actual connection not on the vector

**Review by enen92 — APPROVED** [review]

File: xbmc/network/TCPServer.cpp [hunk 2]
```diff
@@ -227,9 +227,13 @@ void CTCPServer::Announce(ANNOUNCEMENT::AnnouncementFlag flag,
                           const std::string& message,
                           const CVariant& data)
 {
+  size_t connectionsSize = m_connections.size();
```
Comment (enen92) [inline xbmc/network/TCPServer.cpp:230]: actually why do you need to store the vector size?
Probably it's better if you simple use:

```
if (m_connections.empty())
   return;
````
and leave `m_connections.size()` below

**Review by fuzzard — APPROVED** [review]

**Pull request merged**